### PR TITLE
cpp linkage

### PR DESCRIPTION
**Pull request by BrianPugh**

### CODE REVIEW

**Comment by silentbicycle** [comment]
This needs to be targeted at the develop branch, so it can be incorporated into a versioned release.

File: heatshrink_common.h [hunk 1]
```diff
@@ -16,5 +20,9 @@
 
 #define HEATSHRINK_LITERAL_MARKER 0x01
 #define HEATSHRINK_BACKREF_MARKER 0x00
+#ifdef __cplusplus
+
+}
+#endif
```
Comment (BenBE) [inline heatshrink_common.h:26]:
```suggestion
#ifdef __cplusplus
}

#endif
```

File: heatshrink_decoder.h [hunk 2]
```diff
@@ -1,6 +1,10 @@
 #ifndef HEATSHRINK_DECODER_H
 #define HEATSHRINK_DECODER_H
 
+#ifdef __cplusplus
+extern "C"{
```
Comment (BenBE) [inline heatshrink_decoder.h:5]:
```suggestion
extern "C" {
```

File: heatshrink_encoder.h [hunk 3]
```diff
@@ -1,6 +1,10 @@
 #ifndef HEATSHRINK_ENCODER_H
 #define HEATSHRINK_ENCODER_H
 
+#ifdef __cplusplus
+extern "C"{
```
Comment (BenBE) [inline heatshrink_encoder.h:5]:
```suggestion
extern "C" {
```

File: heatshrink_common.h [hunk 4]
```diff
@@ -1,6 +1,10 @@
 #ifndef HEATSHRINK_H
 #define HEATSHRINK_H
 
+#ifdef __cplusplus
+extern "C"{
```
Comment (BenBE) [inline heatshrink_common.h:5]:
```suggestion
extern "C" {
```

**Review by BenBE — COMMENTED** [review]
Just some additional minor notes, when rebasing this onto the `develop` branch (which shouldn't cause any conflicts with these changes) …

**Comment by silentbicycle** [comment]
If I don't hear from you by the time I cut the 0.4.2 release (likely in the next couple days), I will cherry-pick this and credit you in the CHANGELOG.

**Comment by BrianPugh** [comment]
I'll make a new PR based on a branch that's not my master

**Pull request closed without merging**